src/openrct2-ui/windows/TileInspector.cpp

```diff
@@ -510,1888 +466,1767 @@ static uint64_t PageDisabledWidgets[] = {
+            min_height += PageGroupBoxSettings[index].details_top_offset - GROUPBOX_PADDING - 3;
+        }
+        tileInspectorPage = p;
+        auto pageIndex = EnumValue(p);
+        widgets = PageWidgets[pageIndex];
+        hold_down_widgets = PageHoldDownWidgets[pageIndex];
+        disabled_widgets = PageDisabledWidgets[pageIndex];
+        pressed_widgets = 0;
+        InvalidateWindow();
     }
 
-    // Get selected element
-    TileElement* const tileElement = WindowTileInspectorGetSelectedElement(w);
-
-    if (w->tileInspectorPage == TileInspectorPage::Wall)
+    void UpdateSelectedTile(const ScreenCoordsXY& screenCoords)
     {
-        openrct2_assert(tileElement->GetType() == TileElementType::Wall, "Element is not a wall");
+        const bool ctrlIsHeldDown = InputTestPlaceObjectModifier(PLACE_OBJECT_MODIFIER_COPY_Z);
+        // Mouse hasn't moved
+        if (screenCoords.x == _toolMouseX && screenCoords.y == _toolMouseY && _toolCtrlDown == ctrlIsHeldDown)
+            return;
+
+        _toolMouseX = screenCoords.x;
+        _toolMouseY = screenCoords.y;
+        _toolCtrlDown = ctrlIsHeldDown;
 
-        if (widgetIndex == WIDX_WALL_DROPDOWN_SLOPE_BUTTON)
+        CoordsXY mapCoords{};
+        TileElement* clickedElement = nullptr;
+        if (ctrlIsHeldDown)
         {
-            WindowTileInspectorWallSetSlope(windowTileInspectorSelectedIndex, dropdownIndex);
+            auto info = get_map_coordinates_from_pos(screenCoords, ViewportInteractionFlags);
+            clickedElement = info.Element;
+            mapCoords = info.Loc;
         }
-    }
-}
 
-static void WindowTileInspectorToolUpdate(rct_window* w, rct_widgetindex widgetIndex, const ScreenCoordsXY& screenCoords)
-{
-    map_invalidate_selection_rect();
+        // Even if Ctrl was pressed, fall back to normal selection when there was nothing under the cursor
+        if (clickedElement == nullptr)
+        {
+            auto mouseCoords = screen_pos_to_map_pos(screenCoords, nullptr);
+            if (!mouseCoords.has_value())
+                return;
 
-    gMapSelectFlags |= MAP_SELECT_FLAG_ENABLE;
+            mapCoords = mouseCoords.value();
+            // Tile is already selected
+            if (_tileSelected && mapCoords.x == _toolMap.x && mapCoords.y == _toolMap.y)
+                return;
+        }
 
-    CoordsXY mapCoords;
-    TileElement* clickedElement = nullptr;
-    bool mouseOnViewport = false;
-    if (InputTestPlaceObjectModifier(PLACE_OBJECT_MODIFIER_COPY_Z))
-    {
-        auto info = get_map_coordinates_from_pos(screenCoords, ViewportInteractionFlags);
-        clickedElement = info.Element;
-        mapCoords = info.Loc;
+        _tileSelected = true;
+        _toolMap = mapCoords;
+        windowTileInspectorTile = TileCoordsXY(mapCoords);
+        OpenRCT2::TileInspector::SetSelectedElement(clickedElement);
+        LoadTile(clickedElement);
     }
 
```

Comment:
Any reason for removing empty lines like this? There are many more cases like this where they are removed, while they do improve readability.

Reply:
It's been a while, but I think I removed them because they were unused

Reply:
Yes, I mean the blank lines. VS is set up to use clang-format for formatting, perhaps with a newer version it's doing this?